src/Sulu/Bundle/MediaBundle/Controller/CollectionController.php

```diff
@@ -368,14 +369,16 @@ protected function saveEntity($id, Request $request)
             $collectionManager = $this->getCollectionManager();
             $data = $this->getData($request);
             $data['id'] = $id;
-            $data['locale'] = $this->getLocale($request);
+            $data['locale'] = $this->getRequestParameter($request, 'locale', true);
```

Comment:
Why don't you use the `getLocale` method anymore resp. override it to match the collections locales?